### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by carcaschoi** [BreakPointOo/AtmosphereUpdater#1]
剛才在91wii看見你的帖就急不及待想測試看看，先感謝你的分享

剛才我打開了app，我選擇了自己的sd卡(F:)，卻顯示讀取錯誤，請問有沒有辦法解決🙏🙏

![20210501_190328](https://user-images.githubusercontent.com/64573431/116780518-f321a800-aaaf-11eb-9d7d-2d03ecc36dba.jpg)

**Comment by BreakPointOo** [comment]
![image](https://user-images.githubusercontent.com/36473024/116780636-8a86fb00-aab0-11eb-99b5-9ff5f0887800.png)
是這樣的，我將這個軟件定義為升級工具 所以判斷的條件裡會增加一個emuMMC文件夾是否存在 如果不存在該文件夾 則判斷為讀取錯誤...

**Comment by carcaschoi** [comment]
但如果有人沒有emuMMC，那麼這個就不能用了🤔

**Comment by BreakPointOo** [comment]
是的 我剛剛入手switch 並不知道發生這種情況的原因是因為大氣層版本問題還是每個人的整合包文件結構不一樣 如果方便的話可以分享一個你正在使用的整合包下載鏈接嗎

**Comment by carcaschoi** [comment]
我就是自己做ShallowSea整合包的那個人🤣🤣，整合包方面就問對人了😏

我的整合包連結:
https://github.com/carcaschoi/ShallowSea

**Comment by BreakPointOo** [comment]
這太酷了 我應該先修改一下標題 加入‘僅適用於部分整合包’ 關於整合包的內容我還要好好研究下

**Comment by carcaschoi** [comment]
我建議有兩個方案:
1. 如果只是想做Updater的，可以只檢查atmosphere，sept 這兩個文件夾，這兩個是大氣層的核心檔案
2. 可以只檢查sd卡的存在，那麼第一次使用的人也可以直接 使用整合包

**Comment by BreakPointOo** [comment]
感謝反饋  目前的代碼會跳過根目錄下的emuMMC和Nintendo兩個文件夾 修改其他所有文件 所以直接去檢查atmosphere，sept會有破壞emuMMC不在根目錄的整合包文件的風險 所以我覺得對每個風格不同的整合包進行單獨適配比較合適 因此我認為可以加入判斷整合包的代碼 針對整合包進行不同修改

**Comment by carcaschoi** [comment]
哦！我明白了你說什麼，但是這兩個文件夾大氣層核心文件
我先舉例一下吧！
1. sxos用戶: sxos壓根沒有(atmosphere，sept)文件夾，所以馬上會報錯=正常
2. Reinx用戶: 同上
3. 原版大氣層+hekate用戶: 如果虛擬系統不在emuMMC文件夾，就會不能使用emuMMC，所以他們不可能亂放虛擬系統到其他位置

！！！只有自定義修改版大氣層( 例如: https://www.91wii.com/thread-224026-1-2.html )，如果他們修改了虛擬系統讀取位置才會有這個問題， 所以個人建議在提醒畫面是警告他們不要使用自定義(修改版)大氣層

**Comment by BreakPointOo** [comment]
是的！問題的本質就在於虛擬系統是否在根目錄下的emuMMC文件夾，這個工具主要是為了方便新手朋友，使用修改版大氣層的朋友應該不需要這個工具的幫助...
我才剛開始玩破解機，用戶組等級太低了 沒辦法打開這個鏈接...

**Comment by carcaschoi** [comment]
这个版本大气层可以让文件格式虚拟系统的大气和SXOS使用同一个虚拟系统,共用同一个虚拟Nintendo目录,以达到两个系统无缝互切的目的(modules不能共用)。
( ?" V9 |$ k/ ]2 [* p% q- B! K
如果你用的是 隐藏分区 虚拟系统可以直接改emmc目录下配置项目即可 不用下载这个 直接用官方版本大气层

**Comment by carcaschoi** [comment]
> 这个版本大气层可以让文件格式虚拟系统的大气和SXOS使用同一个虚拟系统,共用同一个虚拟Nintendo目录,以达到两个系统无缝互切的目的(modules不能共用)。
> ( ?" V9 |$ k/ ]2 [* p% q- B! K
> 如果你用的是 隐藏分区 虚拟系统可以直接改emmc目录下配置项目即可 不用下载这个 直接用官方版本大气层

我把文章重點copy了出來

**Comment by carcaschoi** [comment]
> 是的！問題的本質就在於虛擬系統是否在根目錄下的emuMMC文件夾，這個工具主要是為了方便新手朋友，使用修改版大氣層的朋友應該不需要這個工具的幫助...
> 我才剛開始玩破解機，用戶組等級太低了 沒辦法打開這個鏈接...

原版hekate造虛擬系統的時候，就是默認在這個emuMMC文件夾構建虛擬系統

**Comment by BreakPointOo** [comment]
原來如此 突然發現後續需要的修改還是蠻多的 當前版本的判斷條件如果去掉emuMMC的話代碼出錯的地方會非常多 有很多的文件我都是在默認存在的情況下修改的
